### ISSUE TITLE
Need to ensure 2fa for GH accounts with access to our build infrastructure

### ISSUE THREAD

**Issue by larsbergstrom** [servo/saltfs#399]
Users with 'r+', operator, or 'try' access should probably be using 2FA, as their accounts have the ability to land changes to Servo and should be protected. These users are listed here:
https://github.com/servo/saltfs/blob/master/homu/files/cfg.toml#L69-L116

One idea is to add them to an organization group ("infra?") and require 2FA for that group.

We should also require this of everyone with write access to the organization, too. That has been granted more widely than is desireable due to the requirement of write/push access in order to change labels.

cc @edunham @aneeshusa

**Comment by aneeshusa** [comment]
AFAIK, the only way to check if a user has 2FA enabled is via the `2fa_disabled` filter to the `/orgs/:org/members` GitHub API endpoint, hence the idea to add r+/try users to a zero-permission group in the servo org and require 2FA. This should be coupled with a Homu check that a user has 2FA enabled (i.e. check that the user is part of the Servo org -> they have 2FA enabled). Ideally, we would check this by hitting the API each time an r+/retry or try comment is posted, but this will likely cause us to hit the rate limit. Instead, we can check membership less frequently (once an hour? once a day?) and cache that value for Homu.

+100 on requiring this for anyone in the organization.

**Comment by aneeshusa** [comment]
How does this interact with [our integration with the Gecko autolander](https://docs.google.com/document/d/1uubYE7JXaVY10PoAY9BVx8A-T11ZxP1RYqNOrFJwdcU/edit?pref=2&pli=1)? I'm particularly interested in this:
```
In the case where #2 is instead a change to servo code is made directly in m-c, a commit of the corresponding changes will be pushed into the GitHub repository
```
This implies that the Gecko autolander has direct push access to the main servo repo; any security policies we set on GitHub accounts also need to be applied to (Bugzilla? not sure) accounts with access to the autolander to be successfully enforced.

**Comment by larsbergstrom** [comment]
Yes, that's correct - any Gecko autolander account will need to have 2fa enabled. We should probably do that to bors-servo, too, though it's not entirely clear to me what the right way to 2fa "robot" accounts is. 

@edunham Do you have any ideas for this one?

**Comment by aneeshusa** [comment]
GitHub now supports [requiring 2fa for all org members](https://help.github.com/articles/requiring-two-factor-authentication-in-your-organization/)!

Before we flip the switch, we need to figure out a story for our bot accounts (bors-servo, possibly others). One option might be the new Integrations feature, which is currently in early-access/beta.
